ConsoleApp/Program.cs

```diff
@@ -0,0 +1,93 @@
+﻿namespace ConsoleApp
+{
+    using System;
+    class ConsoleApp
+    {
+        static int F(int n)
+        {
+            if (n == 0)
+            {
+                return 1;
+            }
+            else
+            {
+                return n - M(F(n - 1));
+            }
+        }
+        static int M(int n)
+        {
+            if (n == 0)
+            {
+                return 0;
+            }
+            else
+            {
+                return n - F(M(n - 1));
+            }
+        }
+        static UInt64 Pow(UInt32 x, UInt32 y)
+        {
+            UInt64 result;
+
+            if(y == 0)
+            {
+                return 1;
+            }
+            if(x == 0)
+            {
+                if (y == 0)
+                {
+                    return 1;
+                }
+                else
+                {
+                    return 0;
+                }
+            }
+
+            result = x;
+            for (int i = 1; i < y; i++)
```

Comment:
the condition was to do it recursively, not in loop

Reply:
The method "Pow" is updated